core/thread/src/TThread.cxx

```diff
@@ -320,7 +320,7 @@ void TThread::Init()
 {
    if (fgThreadImp || fgIsTearDown) return;
 
-#if !defined (_REENTRANT) && !defined (WIN32)
+#if !defined (R__FBSD) || !defined (_REENTRANT) && !defined (WIN32)
```

Comment:
should be
```suggestion
#if !defined (R__FBSD) && !defined (_REENTRANT) && !defined (WIN32)
```
I think (even though it may not be needed at all if you pass `-D_REENTRANT` in `cmake/modules/SetUpFreeBSD.cmake`)